src/prof_sys.c

```diff
@@ -711,26 +723,68 @@ prof_dump_filename(tsd_t *tsd, char *filename, char v, uint64_t vseq) {
 
 	assert(tsd_reentrancy_level_get(tsd) == 0);
 	const char *prefix = prof_prefix_get(tsd_tsdn(tsd));
+#ifdef JEMALLOC_PID_NAMESPACE
+	bool pid_ns_enabled = prof_pid_namespace_enabled(tsd_tsdn(tsd));
+#endif
 
 	if (vseq != VSEQ_INVALID) {
-	        /* "<prefix>.<pid>.<seq>.v<vseq>.heap" */
+#ifdef JEMALLOC_PID_NAMESPACE
```

Comment:
Not super happy with how verbose this is with the ifdefs, would be open to cleaner implementations.